jukebox/src/main/webapp/js/dbScript.js

```diff
@@ -0,0 +1,77 @@
+const NO_VIDEO = -1;
+
+var db = firebase.firestore();
+
+//Adds a song to the queue
+function addSongDB(song_ID, title, room_ID) {
+    db.collection('songs').where("SONG_ID", "==", song_ID)
+    .get()
+    .then(function(querySnapshot) {
+        if(querySnapshot.exists){ //Song is already in database, just add to queue
+            db.collection("queue").doc(song_ID).set( {
```

Comment:
Along the same lines as the comment I made further down, it's not necessary, but we could put the queue collection within the 'rooms' document that it belongs to (rather than using room_id to associate the two collections).